File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
def split_num(n):
    data = []
    num = ''
    for char in n:
        temp = 0
        if char.isdigit():
            num += char
        else:
            if num:
                data.append(num)
                num = ''
            data.append(char)

    if num:
        data.append(num)

    return data


n = input()
dat = split_num(n)
res = int(dat[0])
sw = True

for i in range(1, len(dat)):

    if dat[i] == "-":
        sw = False
    elif sw and dat[i].isdigit():
        res += int(dat[i])

    if not sw and dat[i].isdigit():
        res -= int(dat[i])


print(res)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), "[-+]", true);
        int res = 0;
        boolean state = true;

        while(st.hasMoreTokens()){
            String temp = st.nextToken();
            int num = 0;
            if(temp.equals("-")){
                state = false;
            }
            else if(!temp.equals("+"))
                num = Integer.parseInt(temp);

            res += state? num : num*(-1);
        }
        System.out.println(res);
    }
}
